docs/runtime/instantiable-module.md

```diff
@@ -0,0 +1,140 @@
+
+### Configuration Trait
+```rust
+pub trait Trait<I: Instance>: system::Trait {
+	// TODO: Add other types and constants required configure this module.
+
+	/// The overarching event type.
+	type Event: From<Event<Self, I>> + Into<<Self as system::Trait>::Event>;
+}
+```
+
+### Storage Declaration
+```rust
+decl_storage! {
+	trait Store for Module<T: Trait<I>, I: Instance> as TemplateModule {
+		...
+	}
+}
+```
+
+### Declaring the Module Struct
+```rust
+decl_module! {
+	/// The module declaration.
+	pub struct Module<T: Trait<I>, I: Instance> for enum Call where origin: T::Origin {
+		...
+	}
+}
+```
+### Accessing Storage
+```rust
+<Something<T, I>>::put(something);
+```
+
+<!-- In v2.0 the T is omitted for both instantiable and non-instantiable modules
+<Something<I>>::put(something); -->
+
+### Event initialization
+```rust
+fn deposit_event<T, I>() = default;
+```
+
+### Event Declaration
+```rust
+decl_event!(
+	pub enum Event<T, I> where AccountId = <T as system::Trait>::AccountId {
+		...
+	}
+}
+```
+
+## Deploying an Instantiable Module
+
+When composing a runtime, one must take care to use the proper syntax for instantiable modules. As before it is slightly different.
+
+### Implementing the Configuration Trait
+When implementing the module's configuration trait, you need to be explicit which instance you are configuring, because in general the instances need not have identical configurations. In this snippet we've named the instance `Instance1`.
+```rust
+impl template::Trait<template::Instance1> for Runtime {
+	type Event = Event;
+}
+```
+
+### In the `construct_runtime!` Macro
+Finally, in the `construct_runtime!` macro, we give a more human-friendly name to each instance of the module. Here I've called `Instance1` `FirstTemplate`.
+```rust
+FirstTemplate: template::<Instance1>::{Module, Call, Storage, Event<T>, Config},
+```
+
+The default Substrate node does a similar thing with its council and technical committee.
+```rust
+Council: collective::<Instance1>::{Module, Call, Storage, Origin<T>, Event<T>, Config<T>},
+TechnicalCommittee: collective::<Instance2>::{Module, Call, Storage, Origin<T>, Event<T>, Config<T>}
+```
+
+## Default Instance
+One drawback of instantiable modules, as we've presented them so far is that they require the deployer to use the more elaborate syntax even if they only desire a single instance of the module. To alleviate this inconvenience, Substrate provides a feature known as DefaultInstance. This allows runtime developers to deploy an instantiable module exactly as they would if it were not instantiable provided they **only use a single instance**.
+
+To make your instantiable module support DefaultInstance, you must specify it in three places.
```

Comment:
Why would they not want to support this? Should it just be part of the main instructions for making an instantiable module?

Reply:
> Why would they not want to support this?

I wondered that too. I can't think of a use case.

> Should it just be part of the main instructions for making an instantiable module?

That would make the article shorter, but we would lose the opportunity to explain exactly what DefaultInstance actually does. Personally I prefer it the way it is. One topic at a time.

Reply:
I feel like this is a tangent for someone that has implemented an instantiable module but doesn't want to make it instantiable. It seems unnecessary to cover here; maybe an italicized note somewhere with a link to documentation would suffice like

```
Note: Instantiable modules that contain only a single instance can implement [`DefaultInstance`](INSERT_LINK) for configuration when the runtime is not taking advantage of the module's instancing capabilities.
```